core/modules/replica/ConfigurationIFace.h

```diff
@@ -0,0 +1,1140 @@
+    unsigned int numSubStripes = 0;     /// The number of sub-stripes (from the CSS partitioning configuration)
+    double       overlap = 0.;          /// The default overlap for tables that do not specify their own overlap
+
+    std::shared_ptr<ChunkNumberValidator> chunkNumberValidator;     /// A validator for chunk numbers
+
+    /// @return JSON representation of the object
+    nlohmann::json toJson() const;
+};
+
+std::ostream& operator <<(std::ostream& os, DatabaseFamilyInfo const& info);
+
+
+/**
+  * Class ConfigurationIFace is a very base base class for a family of concrete classes
+  * providing configuration services for the components of the Replication
+  * system.
+  */
+class ConfigurationIFace {
+public:
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<ConfigurationIFace> Ptr;
+
+    // Copy semantics is prohibited
+
+    ConfigurationIFace(ConfigurationIFace const&) = delete;
+    ConfigurationIFace& operator=(ConfigurationIFace const&) = delete;
+
+    virtual ~ConfigurationIFace() = default;
+
+    /// @return the configuration prefix
+    virtual std::string prefix() const = 0;
+
+    /**
+     * Construct the original (minus security-related info) path to
+     * the configuration source.
+     *
+     * @oparam showPassword  if 'false' then hash a password in the result
+     * @return the constructed path
+     */
+    virtual std::string configUrl(bool showPassword=false) const = 0;
+
+    // ------------------------------------------------------------------------
+    // -- Common configuration parameters of both the controller and workers --
+    // ------------------------------------------------------------------------
+
+    /**
+     * @param isEnabled
+     *   select workers which are allowed to participate in the
+     *   replication operations.
+     *
+     * @param isReadOnly
+     *   a subclass of the 'enabled' workers which can only serve as
+     *   a source of replicas. No replica modification (creation or
+     *   deletion) operations would be allowed against those workers.
+     *   NOTE: this filter only matters for the 'enabled' workers.
+     *
+     * @return
+     *   the names of known workers which have the specified properties
+     *   as per input filters.
+     */
+    virtual std::vector<std::string> workers(bool isEnabled=true,
+                                             bool isReadOnly=false) const = 0;
+
+    /// @return names of all known workers regardless of their statuses
+    virtual std::vector<std::string> allWorkers() const = 0;
+
+    /// @return maximum size of the request buffers in bytes
+    virtual size_t requestBufferSizeBytes() const = 0;
+
+    /// @param val  the new value of the parameter
+    virtual void setRequestBufferSizeBytes(size_t val) = 0;
+
+    /// @return timeout in seconds for the network retry operations
+    virtual unsigned int retryTimeoutSec() const = 0;
```

Comment:
This will require too many changes in the client code. I will consider this in some later refactoring effort of the Replication System. By the way, this idea was already registered in my TODO list:
https://confluence.lsstcorp.org/display/DM/Completed%2C+on-going+and+proposed+improvements